test/assert.js

```diff
@@ -375,6 +376,7 @@ throws('should throw if function does not throw Error :: generic', () => {
 throws('should throw if function does not throw matching Error :: RegExp', () => {
 	try {
 		$.throws(() => { throw new Error('hello') }, /world/);
+		assert.unreachable('Function threw correct pattern when it should have thrown incorrect one');
```

Comment:
```suggestion
```

Reply:
(I think this is the only one I’m confused on. Accepted all other changes and happy to go with what you think is best for this.)

Reply:
The assertion helper is saying: given X function, an error (a) should be thrown and (b) the thrown error should match {this condition}. The condition can be defined via a function, a RegExp, or a string.

So this test has X function, which happens to `throw new Error('hello')`, and the assertion is saying X must throw an Error that matches the `/world/` pattern. 

In the original test, this fails (obv because `"hello"` does not match `/world`) and so the assertions within the `catch` block run, and correctly find that the "Expected function to throw exception matching `/world/` pattern" message appears, among others.

When you change the X to be:

```js
$.throws(() => { throw new Error('world') }, /world/);
```

This _doesnt_ fail the assertion, because `"world"` _does_ match the `/world` pattern. The assertion passed. So in this case, the checks within the `catch` block never run.

Testing negated fail conditions can be a trip, haha

Reply:
I kinda feel like it could be helpful to make that work via 
```js
$.throws(() => throw new Error('world') }).matches(new Error('world'));
```
which maybe that doesn't make sense.. but it'd be nice to be able to chain `throws()` with `match()` rather than having a regex over `err.message` 🤔